keps/2724-topology-aware-scheduling/README.md

```diff
@@ -743,6 +747,67 @@ to ungate a Pod. The expectation is fulfilled if the Pod is observed as ungated
 or the ungating request fails. We hold ungating if there are pending ungatings
 within the PodSet.
 
+### Support for ProvisioningRequests
+
+We are going to support autoscaling via ProvisioningRequest AdmissionCheck.
+
+The key assumptions of the solution:
+- When there is ProvisioningRequest on the list of AdmissionChecks Kueue
+  Scheduler only reserves quota, but skips TAS assignment until all
+  AdmissionChecks (including the ProvisioningRequest) are `Ready` (green).
+- When all AdmissionChecks are `Ready` Kueue Scheduler does "second pass" on the
+  workload to extend the "TAS" assignment, but respecting the previously
+  selected resource flavor.
+- In order to do "second pass" for such workloads Kueue Scheduler maintains a
```

Comment:
Have we thought of adding some metrics here to track this behavior? I was thinking of adding something like
`workload_second_pass_retries_total` (Tracks retry attempts for second-pass failures to identify scheduling bottlenecks) and `pending_second_pass_workloads` (Counts workloads in the `secondPassQueue`).

Reply:
Good idea, I was so focused on the prototype that haven't yet thought about metrics, but both `pending_second_pass_workloads` and `workload_second_pass_retries_total` sound useful. We should in general add some metrics to TAS, but it was so far neglected to get it to MVP state.